public/utils/helpers.ts

```diff
@@ -45,3 +49,102 @@ export function diffJson(oldJson?: Record<string, any>, newJson?: Record<string,
   }
   return initial + addOrChanged + oldKeys.length;
 }
+
+// code related to filter blocked index/alias/datastream
+// an example to use:
+// import { aliasBlockedPredicate, filterBlockedItems } from "./helpers";
+// import { IndexOpBlocksType } from "./constants";
+// const result = filterBlockedItems<IAlias>(services, selectedItems, IndexOpBlocksType.Closed, aliasBlockedPredicate)
+
+interface BlockedIndices {
+  [indexName: string]: String[];
+}
+
+interface ClusterBlocksStateResponse {
+  blocks: {
+    indices?: {
+      [indexName: string]: {
+        [blockId: string]: {
+          description: string;
+          retryable: boolean;
+          levels: string[];
+        };
+      };
+    };
+  };
+}
+
+export async function getBlockedIndices(broswerServices: BrowserServices): Promise<BlockedIndices> {
+  const result = await broswerServices.commonService.apiCaller<ClusterBlocksStateResponse>({
+    endpoint: "cluster.state",
+    data: {
+      metric: "blocks",
+    },
+  });
+  if (!result.ok) {
+    throw result.error;
+  }
+
+  const blocksResponse = result.response.blocks;
+  if (!blocksResponse.indices) {
+    return {};
+  }
+
+  const blockedIndices: BlockedIndices = {};
+  Object.keys(blocksResponse.indices).forEach((indexName) => {
+    const innerBlocksInfo = blocksResponse.indices![indexName];
+    const indexOpBlocksIds = Object.keys(innerBlocksInfo);
+    blockedIndices[indexName] = indexOpBlocksIds;
+  });
+  return blockedIndices;
+}
+
+export interface FilteredBlockedItems<T> {
+  unBlockedItems: T[];
+  blockedItems: T[];
+}
+
+export function indexBlockedPredicate(item: CatIndex, blockedItemsSet: Set<string>): boolean {
+  return blockedItemsSet.has(item.index);
+}
+
+export function aliasBlockedPredicate(item: IAlias, blockedItemsSet: Set<String>): boolean {
+  return !!item.indexArray.find((indexName) => blockedItemsSet.has(indexName));
+}
+
+export function dataStreamBlockedPredicate(item: DataStream, blockedItemsSet: Set<String>): boolean {
+  return !!item.indices.find((dataStreamIndex) => blockedItemsSet.has(dataStreamIndex.index_name));
+}
+
+export async function filterBlockedItems<T>(
```

Comment:
I'd recommend making this function as pure filter function, without the async request to get blocked indexes. The benefit will be that the filter function can be used in other cases, not only with the cluster.state API. But you can make the call.